Framework/Algorithms/src/Stitch1DMany.cpp

```diff
@@ -113,151 +176,248 @@ std::map<std::string, std::string> Stitch1DMany::validateInputs() {
     errors["InputWorkspaces"] = "Input workspaces must be given";
   }
 
-  m_numWorkspaces = m_inputWorkspaces.size();
-
   m_startOverlaps = this->getProperty("StartOverlaps");
   m_endOverlaps = this->getProperty("EndOverlaps");
+  m_scaleRHSWorkspace = this->getProperty("ScaleRHSWorkspace");
+  m_params = this->getProperty("Params");
 
-  if (!m_startOverlaps.empty() && m_startOverlaps.size() != m_numWorkspaces - 1)
+  m_numWSPerGroup = m_inputWSMatrix[0].size();
+  m_numWSPerPeriod = m_inputWSMatrix.size();
+
+  size_t numStitchableWS =
+      (m_numWSPerPeriod > 1) ? m_numWSPerPeriod : m_numWSPerGroup;
+
+  if (!m_startOverlaps.empty() && m_startOverlaps.size() != numStitchableWS - 1)
     errors["StartOverlaps"] = "If given, StartOverlaps must have one fewer "
                               "entries than the number of input workspaces.";
 
   if (m_startOverlaps.size() != m_endOverlaps.size())
     errors["EndOverlaps"] =
         "EndOverlaps must have the same number of entries as StartOverlaps.";
 
-  m_scaleRHSWorkspace = this->getProperty("ScaleRHSWorkspace");
-  m_useManualScaleFactor = this->getProperty("UseManualScaleFactor");
-  m_manualScaleFactor = this->getProperty("ManualScaleFactor");
-  m_params = this->getProperty("Params");
-
-  if (m_params.empty())
-    errors["Params"] = "At least one parameter must be given.";
-
   if (!m_scaleRHSWorkspace) {
     // Flip these around for processing
-    std::reverse(m_inputWorkspaces.begin(), m_inputWorkspaces.end());
     std::reverse(m_startOverlaps.begin(), m_startOverlaps.end());
     std::reverse(m_endOverlaps.begin(), m_endOverlaps.end());
+    for (auto &period : m_inputWSMatrix) {
+      std::reverse(period.begin(), period.end());
+    }
   }
```

Comment:
Seems that there's no need to do the above `if` (see my other PR). Can you remove it and check the effect this has? I think with this `if` the algorithm fails if `ScaleRHSWorkspace = false`, without it, it should work and results should make sense. Could you check this?